code/game/gamemodes/objective_holder.dm

```diff
@@ -106,7 +106,7 @@
 /datum/objective_holder/proc/handle_objective(datum/objective/Objective)
 	for(var/loop in 1 to 5)
 		Objective.find_target(assigned_targets)
-		if(Objective.found_target()) // handles normal objectives, and steal objectives
+		if(Objective.found_target() || Objective.delayed_objective) // Handles normal objectives, and steal objectives
```

Comment:
Why not put Objective.delayed_objective before this loop? It won't matter if the objective is rerolled anyways

Reply:
Done!